climada/engine/uncertainty/unc_cost_benefit.py

```diff
@@ -0,0 +1,227 @@
+        ----------
+        haz_unc : climada.engine.uncertainty.UncVar
+                  or climada.hazard.Hazard
+            Hazard uncertainty variable or Hazard for the present Hazard
+            in climada.engine.CostBenefit.calc
+        ent_unc : climada.engine.uncertainty.UncVar
+                  or climada.entity.Entity
+            Entity uncertainty variable or Entity for the future Entity
+            in climada.engine.CostBenefit.calc
+        haz_unc_fut: climada.engine.uncertainty.UncVar
+                     or climada.hazard.Hazard, optional
+            Hazard uncertainty variable or Hazard for the future Hazard
+            in climada.engine.CostBenefit.calc
+            The Default is None.
+        ent_fut_unc : climada.engine.uncertainty.UncVar
+                      or climada.entity.Entity, optional
+            Entity uncertainty variable or Entity for the future Entity
+            in climada.engine.CostBenefit.calc
+
+        """
+
+        unc_vars = {
+            'haz': UncVar.var_to_uncvar(haz_unc),
+            'ent': UncVar.var_to_uncvar(ent_unc),
+            'haz_fut': UncVar.var_to_uncvar(haz_fut_unc),
+            'ent_fut': UncVar.var_to_uncvar(ent_fut_unc)
+            }
+
+        metrics =  {
+            'tot_climate_risk': pd.DataFrame([]),
+            'benefit': pd.DataFrame([]),
+            'cost_ben_ratio': pd.DataFrame([]),
+            'imp_meas_present': pd.DataFrame([]),
+            'imp_meas_future': pd.DataFrame([])
+            }
+
+        Uncertainty.__init__(self, unc_vars=unc_vars, metrics=metrics)
+
+
+    def calc_distribution(self, pool=None, **kwargs):
+        """
+        Computes the cost benefit for each of the parameters set defined in
+        uncertainty.samples.
+
+        By default, imp_meas_present, imp_meas_future, tot_climate_risk,
+        benefit, cost_ben_ratio are computed.
+
+        Parameters
+        ----------
+        pool : pathos.pools.ProcessPool, optional
+            Pool of CPUs for parralel computations. Default is None.
+            The default is None.
+        **kwargs : keyword arguments
+            Any keyword arguments of climada.engine.CostBenefit.calc()
+            EXCEPT: haz, ent, haz_fut, ent_fut
+
+        Returns
+        -------
+        None.
+
+        """
+
+        if self.samples_df.empty:
+            raise ValueError("No sample was found. Please create one first" + 
+                        "using UncImpact.make_sample(N)")
+            
+        start = time.time()
+        one_sample = self.samples_df.iloc[0:1].iterrows()
+        cb_metrics = map(self._map_costben_calc, one_sample)
+        [imp_meas_present,
+         imp_meas_future,
+         tot_climate_risk,
+         benefit,
+         cost_ben_ratio] = list(zip(*cb_metrics))
+        elapsed_time = (time.time() - start) 
+        est_com_time = self.est_comp_time(elapsed_time, pool)
+        LOGGER.info(f"\n\nEstimated computation time: {est_com_time}s\n")
```

Comment:
Why do you need the one_sample evaluation (block 118-127). Aren't you evaluating the full sample set just below - and also overwriting the 5 outputs? (Same in the Impact Uncertainty code)

Reply:
The idea is to get an estimate of the computation time by computing the output for one single sample, and then multiply by the number of samples.

Reply:
Mh ok.. why don't you just provide the time it takes to evaluate the full set? You could 1) drop those few lines, 2) provide the exact computation as it may not scale linearly, 3) you need to compute it ayway...

Reply:
Well, that is then a very different use case. The goal here is to warn the user if the computation is longer than usual. So far, every person that used this model made very simple basic mistakes that changed the computation time from a few seconds/minutes to several hours/days. Hence, my idea to always give a quick estimate of the total computation time (it is a very rough one, but enough to catch these very very bad mistakes), and suggest one of the major problems I spotted so far. 

If someone wants to do a scaling analysis, they should do it themselves.

Reply:
Ok now it makes more sense..

Reply:
If you have a more elegant solution to that, please let me know!

Reply:
if your goal is to warn the user about the total computation time in order for her to quit the process in case it takes too long.. then I think it makes sense to do it the way you do it... I just did not see the point but now I do